tests/cilium_tls_tcp_integration_test.cc

```diff
@@ -146,13 +147,15 @@ class CiliumTLSIntegrationTest : public CiliumTcpIntegrationTest {
 
     auto cfg_or_error = Extensions::TransportSockets::Tls::ServerContextConfigImpl::create(
```

Comment:
in this file i get an additional lint issue, maybe worth to fix?

```
payload_reader_.reset(p: new WaitForPayloadReader(&dispatcher: *dispatcher_));     ● clang-tidy: Use std::make_shared instead (fix available)
```

Reply:
`reset` only takes bare pointers, so maybe the proposed fix is to use assignment instead? I'll leave this for later, unless you have `make tidy` complaining about this?

Reply:
Using make_shared instead as suggested